lib/mysql/mysql.go

```diff
@@ -612,31 +614,63 @@ func (c *Connection) decodePacket(body []byte) (PacketInfo, error) {
 	}
 }
 
+// if n, _ := io.Read(body), returns a hex representation of body[:n] that is at most 96 bytes long.
+func trunc(body []byte, n int) string {
+	if body == nil {
+		return "<nil>"
+	}
+	if n > len(body) {
+		n = len(body)
+	}
+	if n < 1 {
+		return "<empty>"
+	}
+	if n < 48 {
+		return fmt.Sprintf("%x", body[:n])
+	}
+	return fmt.Sprintf("%x...[%d bytes]...%x", body[:16], n - 32, body[n-16:])
+}
```

Comment:
- [ ] Clarify truncation

Reply:
So you don't think it's worth it to add an explicit slice at the end? My concern is that the math is kind of fragile and if someone tweaks it, it could end up coming out to longer than 96 bytes. Seems easy to at the very end just return a slice that is max len 96..

Reply:
It looks like this comment is pinned to the old revision? I added [this](https://github.com/zmap/zgrab2/blob/jb/mysql-timeout/lib/mysql/mysql.go#L621) to ensure that it never goes over, even if we do somehow get a 100MB size.

Reply:
But...d'oh. Probably should only do that if `len(result) > 96`.

Reply:
> It looks like this comment is pinned to the old revision? I added this to ensure that it never goes over, even if we do somehow get a 100MB size.

Oh that looks great, sorry. I think I returned to a window without seeing the refresh link at the top of the page. Thanks for the edits!